cpp/src/arrow/io/caching.cc

```diff
@@ -188,7 +188,9 @@ struct ReadRangeCache::Impl {
       entries = std::move(new_entries);
     }
     // Prefetch immediately, regardless of executor availability, if possible
-    return file->WillNeed(ranges);
+    // As this is optimisation only, failures should not be treated as fatal
+    ARROW_UNUSED(file->WillNeed(ranges));
```

Comment:
I would rather we just ignore IOError but less other errors propagate (they might be indicative of a logic error).

Reply:
Gotcha, have updated it now.